test_requests.py

```diff
@@ -935,6 +935,19 @@ def test_can_send_nonstring_objects_with_files(self):
 
         assert 'multipart/form-data' in p.headers['Content-Type']
 
+    def test_can_send_bytes_bytearray_objects_with_files(self):
+        # Test bytes:
+        data = {'a': 0.0}
```

Comment:
@Lukasa I thought we expected all data to be bytes/string and were generally against coercing int, float, etc.?